library/src/blas1/rocblas_swap_strided_batched.cpp

```diff
@@ -0,0 +1,164 @@
+    template <>
+    constexpr char rocblas_swap_strided_batched_name<rocblas_float_complex>[]
+        = "rocblas_cswap_strided_batched";
+    template <>
+    constexpr char rocblas_swap_strided_batched_name<rocblas_double_complex>[]
+        = "rocblas_zswap_strided_batched";
+
+    template <typename T>
+    rocblas_status rocblas_swap_strided_batched_impl(rocblas_handle handle,
+                                                     rocblas_int    n,
+                                                     T*             x,
+                                                     rocblas_int    incx,
+                                                     rocblas_int    stridex,
+                                                     T*             y,
+                                                     rocblas_int    incy,
+                                                     rocblas_int    stridey,
+                                                     rocblas_int    batch_count)
+    {
+        if(!handle)
+            return rocblas_status_invalid_handle;
+
+        auto layer_mode = handle->layer_mode;
+        if(layer_mode & rocblas_layer_mode_log_trace)
+            log_trace(handle,
+                      rocblas_swap_strided_batched_name<T>,
+                      n,
+                      x,
+                      incx,
+                      stridex,
+                      y,
+                      incy,
+                      stridey,
+                      batch_count);
+        if(layer_mode & rocblas_layer_mode_log_bench)
+            log_bench(handle,
+                      "./rocblas-bench -f swap_strided_batched -r",
+                      rocblas_precision_string<T>,
+                      "-n",
+                      n,
+                      "--incx",
+                      incx,
+                      "--incy",
+                      incy,
+                      "--stride_x",
+                      stridex,
+                      "--stride_y",
+                      stridey,
+                      "--batch",
+                      batch_count);
+        if(layer_mode & rocblas_layer_mode_log_profile)
+            log_profile(handle,
+                        rocblas_swap_strided_batched_name<T>,
+                        "N",
+                        n,
+                        "incx",
+                        incx,
+                        "stride_x",
+                        stridex,
+                        "incy",
+                        incy,
+                        "stride_y",
+                        stridey,
+                        "batch",
+                        batch_count);
+
+        if(!x || !y)
+            return rocblas_status_invalid_pointer;
+
+        size_t abs_incx = incx >= 0 ? incx : -incx;
+        size_t abs_incy = incy >= 0 ? incy : -incy;
+        if(!incx || !incy || (stridex < n * abs_incx) || (stridey < n * abs_incy) || stridex < 0
```

Comment:
Not sure if we've discussed whether or not incx/incy == 0 is an invalid size or not. Does this follow the same logic as negative n? Note the non batched version and Fortran does not explicitly handle this case.

Reply:
> Not sure if we've discussed whether or not incx/incy == 0 is an invalid size or not. Does this follow the same logic as negative n? Note the non batched version and Fortran does not explicitly handle this case.

`n <= 0` causes swap to return immediately with no operation. It is not an error.

`incx == 0` and `incy == 0` are accepted by the original BLAS, but are useless for `SWAP` since they repeatedly swap the same first element of `X` or `Y`.

I say we should not try to detect and flag `incx == 0` or `incy == 0` in `SWAP`.

In other routines like `DOT`, a zero increment would multiply the first element of one vector by another.

It's a degenerate corner case but we don't need to detect and flag it.

Reply:
removed all invalid parameter guards, should match '78 checking but 0 increment is non-deterministic and won't be unit tested